Makefile

```diff
@@ -981,8 +981,14 @@ test-e2e: $(GINKGO) generate-e2e-templates ## Run the end-to-end tests
 kind-cluster: ## Create a new kind cluster designed for development with Tilt
 	hack/kind-install-for-capd.sh
 
+.PHONY: build-kind-test-images
+build-kind-test-images: ## Build the corresponding kindest/node images required for e2e testing
+	scripts/ci-build-kind.sh
+
 .PHONY: tilt-up
-tilt-up: kind-cluster ## Start tilt and build kind cluster if needed.
+tilt-up: kind-cluster ## Start tilt, build kind cluster, generate e2e templates and kindest/node images if needed.
+	$(MAKE) build-kind-test-images
```

Comment:
+1 for having that make target, but I think we should not build all kind images just when running on tilt. I may not use any of the built images 🤔

Reply:
right yeah makes sense -- although i think these functions are "smart" enough already to only build the images if they arent available: https://github.com/kubernetes-sigs/cluster-api/blob/b29e26c0210d12fffb1c1d59e5bbcd492242801e/scripts/ci-e2e-lib.sh#L159

or am i misunderstanding your point?

Reply:
It is not necessary to build images when running tilt (Tilt itself take care of building images)